### ISSUE TITLE
Regression in 3.13 (pngload_buffer)?

### ISSUE THREAD

**Issue by bsboris** [imgproxy/imgproxy#1086]
Hi! It looks like there is a regression in latest build (Imgproxy Pro). We're experiencing following error while applying PNG watermarks to JPEG images:

```
ERROR   [2023-01-16T14:56:49Z] Completed in 744.811501ms some-url.jpg method=GET status=500 client_ip=127.0.0.1 error="pngload_buffer: out of order read at line 600\n"
/app/vips/vips.go:882 github.com/imgproxy/imgproxy/v3/vips.(*Image).CopyMemory
/app/processing/watermark.go:111 github.com/imgproxy/imgproxy/v3/processing.applyWatermark
/app/processing/watermark.go:126 github.com/imgproxy/imgproxy/v3/processing.watermark
/app/processing/pipeline.go:58 github.com/imgproxy/imgproxy/v3/processing.pipeline.Run
/app/processing/processing.go:460 github.com/imgproxy/imgproxy/v3/processing.ProcessImage
/app/processing_handler.go:428 main.handleProcessing.func3
/app/processing_handler.go:426 main.handleProcessing
/app/server.go:137 main.withCORS.func1
/app/server.go:113 main.withDevRequestLimit.func1
/app/server.go:208 main.withPanicHandler.func1
```

It happens randomly and mostly for larger file sizes. Reproducible for various source and watermark images.

**Comment by DarthSim** [comment]
Hey @bsboris!
Could you show the processing options you used? It'd narrow the search field.

**Comment by bsboris** [comment]
Sure, here it is:
`rs:fit:1280:0:0/wm:1.0:re:0:0:0.6/wmu:aHR0cDovL2Rpc2sud2ZvbGlvLnJ1L2Rpc2tzLzMvcHJvamVjdHMvd2F0ZXJtYXJrcy8xNjczODc5MTAwXzgxZWNiMC5wbmc/cb:v2/aHR0cDovL2Rpc2su/d2ZvbGlvLnJ1L2Rp/c2tzLzMvcHJvamVj/dHMvMTkxNjU4NC9w/cmV2aWV3cy8xNjcz/NTA0MTczXzZlNzE3/ZC5qcGc.jpg`

Let me know if you need anything else.

**Comment by DarthSim** [comment]
Thanks for the info, I managed to replicate the issue.  Should be fixed in `v3.13.1`
